uk.co.compucorp.civicrm.hrleaveandabsences/tests/phpunit/CRM/HRLeaveAndAbsences/Service/PublicHolidayLeaveRequestDeletionTest.php

```diff
@@ -648,16 +710,16 @@ public function testCanDeleteAllPublicHolidayLeaveRequestsForAbsencePeriodForSpe
     $this->fabricatePublicHolidayLeaveRequestWithMockBalanceChange($contact['id'], $publicHoliday);
     $this->fabricatePublicHolidayLeaveRequestWithMockBalanceChange($contact2['id'], $publicHoliday);
 
-    $this->assertNotNull(LeaveRequest::findPublicHolidayLeaveRequest($contact['id'], $publicHoliday));
-    $this->assertNotNull(LeaveRequest::findPublicHolidayLeaveRequest($contact2['id'], $publicHoliday));
+    $this->assertNotNull(LeaveRequest::findPublicHolidayLeaveRequest($contact['id'], $publicHoliday, $this->absenceType));
+    $this->assertNotNull(LeaveRequest::findPublicHolidayLeaveRequest($contact2['id'], $publicHoliday, $this->absenceType));
 
     //Delete all Public holiday leave requests for absence period for first contact only
     $deletionLogic = new PublicHolidayLeaveRequestDeletion(new JobContractService());
     $deletionLogic->deleteAllForAbsencePeriod($period, [$contact['id']]);
 
     //Leave Request for PublicHoliday will be deleted for contact1 only
-    $this->assertNull(LeaveRequest::findPublicHolidayLeaveRequest($contact['id'], $publicHoliday));
+    $this->assertNull(LeaveRequest::findPublicHolidayLeaveRequest($contact['id'], $publicHoliday, $this->absenceType));
 
-    $this->assertNotNull(LeaveRequest::findPublicHolidayLeaveRequest($contact2['id'], $publicHoliday));
+    $this->assertNotNull(LeaveRequest::findPublicHolidayLeaveRequest($contact2['id'], $publicHoliday, $this->absenceType));
```

Comment:
A comment not about this specific test, but about all the tests in general: I can see you've updated quite a lot of tests to make sure they work fine with multiple absence types. I wonder if it wouldn't be simpler to have separate tests for these new scenarios and only do that where multiple MTPHL is really relevant (that is, if there's already another test checking the multiple absence type logic, do we really need to repeat that in other places?)

Reply:
I think we have quite a lot of tests already and I didn't want to add more than necessary, hence the approach to modify some of the tests to use multiple absence types. I only modified some that are necessary as there are still quite a LOT more that use single absence types.

The aim was just to modify some of the tests mainly tests that are responsible for creating public holiday requests via the various entry points like work pattern, absence period and so on.
If we add new tests for all these scenarios it will be almost a repetition of the tests that use singular absence types as the only difference will be that one test use one absence type while the other uses multiple types and the aim of both variant of tests is just to demonstrate that public holiday leave requests will be created for as many absence type that has MTPHL.